lib/improver/tests/utilities/test_mathematical_operations.py

```diff
@@ -0,0 +1,466 @@
+        self.positive_integrated_cube.data = (
+            np.zeros(self.positive_integrated_cube.shape))
+
+        # Cubes for integrating in the negative direction.
+        new_cube = cube.copy()
+        # Sort cube so that it is in the expected order.
+        index = [[2, 1, 0], slice(None), slice(None), slice(None)]
+        new_cube = new_cube[tuple(index)]
+        self.negative_upper_bounds_cube = new_cube[:-1, ...]
+        self.negative_lower_bounds_cube = new_cube[1:, ...]
+        self.negative_integrated_cube = new_cube[1:, ...]
+        self.negative_integrated_cube.data = (
+            np.zeros(self.negative_integrated_cube.shape))
+
+    def test_basic(self):
+        """Test that a cube is returned by the perform_integration method with
+        the expected coordinate points."""
+        coord_name = "height"
+        direction = "negative"
+        result = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).perform_integration(
+                      self.negative_upper_bounds_cube,
+                      self.negative_lower_bounds_cube,
+                      self.negative_integrated_cube))
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            result.coord("height").points, np.array([5., 10.]))
+
+    def test_data(self):
+        """Test that the resulting cube contains the expected data following
+        vertical integration."""
+        expected = np.array(
+            [[[[45.00, 32.50, 32.50],
+               [32.50, 32.50, 32.50],
+               [32.50, 32.50, 32.50]]],
+             [[[25.00, 25.00, 25.00],
+               [25.00, 25.00, 25.00],
+               [25.00, 25.00, 25.00]]]])
+        coord_name = "height"
+        direction = "negative"
+        result = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).perform_integration(
+                      self.negative_upper_bounds_cube,
+                      self.negative_lower_bounds_cube,
+                      self.negative_integrated_cube))
+        self.assertArrayAlmostEqual(
+            result.coord("height").points, np.array([5., 10.]))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_start_point_positive(self):
+        """Test that the resulting cube contains the expected data when a
+        start_point is specified, so that only part of the column is
+        integrated. For integration in the positive direction (equivalent to
+        integrating downwards for a height coordinate), the presense of a
+        start_point indicates that the integration may start above the
+        lowest height within the column to be integrated."""
+        expected = np.array(
+            [[[25.00, 25.00, 25.00],
+              [25.00, 25.00, 25.00],
+              [25.00, 25.00, 25.00]]])
+        coord_name = "height"
+        start_point = 8.
```

Comment:
If the start point is such that no layers are found then it returns the rather useless error message
 `ValueError: can't merge an empty CubeList`
It would be better if we caught this error. This is also the same for end_point

Reply:
I'm now catching the exception, in the case that no integration can be done, and raising an error.